x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview_header/engine_overview_header.test.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { EngineOverviewHeader } from '../engine_overview_header';
+import { KibanaContext } from '../../..';
+
+describe('EngineOverviewHeader', () => {
+  let enterpriseSearchUrl;
+
+  afterEach(() => {
+    enterpriseSearchUrl = undefined;
+  });
+
+  const render = () => {
+    return mount(
+      <KibanaContext.Provider
+        value={{
+          http: {},
+          enterpriseSearchUrl,
+          setBreadcrumbs: jest.fn(),
+        }}
+      >
+        <EngineOverviewHeader />
+      </KibanaContext.Provider>
+    );
+  };
+
+  describe('when enterpriseSearchUrl is set', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      enterpriseSearchUrl = 'http://localhost:3002';
+      wrapper = render();
+    });
+
+    describe('the Launch App Search button', () => {
+      const subject = () => wrapper.find('EuiButton[data-test-subj="launch_button"]');
+
+      it('should not be disabled', () => {
+        expect(subject().props().isDisabled).toBeFalsy();
+      });
+
+      it('should use the enterpriseSearchUrl as the base path for its href', () => {
+        expect(subject().props().href).toBe('http://localhost:3002/as');
+      });
+    });
+  });
+
+  describe('when enterpriseSearchUrl is not set', () => {
```

Comment:
From an end user perspective, what circumstances would cause KibanaContext to not provide our plugin with an enterpriseSearchUrl? I'm trying to understand better if disabling the Launch App Search button is the appropriate behavior here.

Reply:
enterpriseSearchUrl is a configuration provided in a config file, by the user.

Reply:
👍